kratos/input_output/vtk_output.cpp

```diff
@@ -0,0 +1,452 @@
+        if( cell_type < 0)
+            KRATOS_ERROR<<"Modelpart contains elements with geometries for which no VTK-output is implemented!"<<std::endl;
+
+        WriteScalarDataToFile( (int)cell_type, rFileStream);
+        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
+    }
+
+    // write conditions types
+    for (const auto& condition : local_mesh.Conditions())
+    {
+        int  cell_type = -1;
+
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle2D3)
+             cell_type = 5; // triangle 2d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4)
+             cell_type = 9; // quad 2d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4)
+             cell_type = 10; // tetra 3d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line2D2)
+             cell_type = 3; // line 2d or 3d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line3D2)
+             cell_type = 3; // line 2d or 3d
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point2D)
+             cell_type = 1;
+        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point3D)
+             cell_type = 1;
+        if( cell_type < 0)
+            KRATOS_ERROR<<"Modelpart contains elements with geometries for which no VTK-output is implemented!"<<std::endl;
+
+        WriteScalarDataToFile( (int)cell_type, rFileStream);
+        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
+    }
+
+
+}
+
+void VtkOutput::WriteNodalResultsAsPointData(const ModelPart &rModelPart, std::ofstream& rFileStream)
+{
+    const auto& local_mesh = rModelPart.GetCommunicator().LocalMesh();
+    // write nodal results header
+    Parameters nodal_results = this->mrOutputSettings["nodal_solution_step_data_variables"];
+    rFileStream << "POINT_DATA " << local_mesh.NumberOfNodes() << "\n";
+
+    for (unsigned int entry = 0; entry < nodal_results.size(); entry++)
+    {
+        // write nodal results variable header
+        std::string nodal_result_name = nodal_results[entry].GetString();
+         VtkOutput::WriteDataType data_characteristic; // 0: unknown, 1: Scalar value, 2: 3 component vector
+        if (KratosComponents<Variable<double>>::Has(nodal_result_name))
+        {
+            data_characteristic =  VtkOutput::WriteDataType::VTK_SCALAR;
+            rFileStream << "SCALARS " << nodal_result_name << " double"
+                       << " 1"
+                       << "\n";
+            rFileStream << "LOOKUP_TABLE default"
+                       << "\n";
+        }
+        else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(nodal_result_name))
```

Comment:
would paraview also support Matrix?
If so then you should probably throw a warning for unsupported var-types